client/reader/stream/index.jsx

```diff
@@ -613,6 +636,18 @@ class ReaderStream extends Component {
 			isRequesting = true;
 		}
 
+		// Due to infinite scrolling or fast selection, API call responses may include items from previously selected feeds.
+		// To temporarily address this issue, filtering out items that don't belong to the currently selected feed.
+		//
+		// The proper fix here is to cancel the pending API calls after the feed selection is changed.
```

Comment:
The current workaround successfully filters content when a specific feed is chosen, but the problem remains when rapidly switching between feeds and ultimately selecting the "All" option.

Are there any ideas on how to properly resolve this issue? I've tried using `AbortController`, but the implementation was taking time. For now I've put that approach on hold and would appreciate feedback if any.

Reply:
Ah, yeah that is going to happen since we're sharing the same `following` stream state in Redux. You're right in that I think the right fix is to cancel pending requests each time a new feed is selected in the sidebar. This isn't ideal but I don't really know of another fix with this approach.

@allilevine @DustyReagan  any thoughts here?

Reply:
I found two alternatives:

1. Assign an id (e.g. a timestamp) to each request in `fetchNextPage`, and then check if we're on the current request when receiving the data. It doesn't cancel the pending request, but it doesn't return the data from the old request.
2. Do something similar with a Redux thunk that "cancels" the older request.

Neither completely cancels the network request like `AbortController` though.

Reply:
I will look into this in a separate PR. Let me know if there are any concerns.

Reply:
> Here's the [branch](https://github.com/Automattic/wp-calypso/tree/protoype/test-using-feed-endpoint) if you want to take a look. It's just prototype code but might be a direction we could take.

@holdercp, thank you for investigating this issue. I've found that we are using a same pattern in other parts of the  codebase

Addressed the feedback in 06176c57f08039d99e827af00264b10b076d16cc and removed my temporary fix.